packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -1163,10 +1139,15 @@ class _ReorderableItemState extends State<_ReorderableItem> {
     return _targetOffset;
   }
 
-  void updateForGap(int gapIndex, double gapExtent, bool animate, bool reverse) {
-    final Offset newTargetOffset = (gapIndex <= index)
-        ? _extentOffset(reverse ? -gapExtent : gapExtent, _listState._scrollDirection)
-        : Offset.zero;
+  void updateForGap(int dragIndex, int gapIndex, double gapExtent, bool animate, bool reverse) {
+    late final Offset newTargetOffset;
+    if (gapIndex < dragIndex && index < dragIndex && index >= gapIndex) {
+      newTargetOffset = _extentOffset(reverse ? -gapExtent : gapExtent, _listState._scrollDirection);
+    } else if (gapIndex > dragIndex && index > dragIndex && index < gapIndex) {
+      newTargetOffset = _extentOffset(reverse ? gapExtent : -gapExtent, _listState._scrollDirection);
+    } else {
+      newTargetOffset = Offset.zero;
+    }
```

Comment:
This is a bit hard to follow, can you split up some of the logic, or add comments to help understand what is happening here?